src/Field/Square.java

```diff
@@ -1,8 +1,27 @@
 package Field;
 
-public interface Square {
+public abstract class Square {
 
-    boolean isAMine();
+    private boolean isRevealed = false;
+    private int hintCount = 0;
 
-    boolean isRevealed();
+    boolean isAMine() {
```

Comment:
I like what you've done with this class by making it abstract. However, just because it's an abstract class doesn't mean you have to implement everything. `isAMine()` is an example of a method I wouldn't implement and would leave to the inherited classes to implement like you had before. I think it's confusing for a generic 'Square' to not be a mine. Would make more sense to leave this blank, and then set this to false in `SafeSquare`

Reply:
I agree, it doesn't make sense how I had it. I have fixed this now.